src/libstd/panicking.rs

```diff
@@ -149,6 +149,18 @@ impl<'a> PanicInfo<'a> {
         self.payload
     }
 
+    /// Returns the payload as a slice if it is a `&'static str` or `String`.
```

Comment:
Not sure, but better safe than sorry! It could also be that Any eventually becomes able to downcast to trait objects for traits that the type itself implements or something like that, where we'd start downcasting to `&Display`, though I guess in that case this signature wouldn't work anymore.

Thoughts, @alexcrichton?